ddtrace/contrib/sqlalchemy/engine.py

```diff
@@ -47,33 +63,61 @@ def __init__(self, tracer, service, engine):
             app=self.vendor,
             app_type=sqlx.APP_TYPE)
 
+        # attach the PIN
+        Pin(
+            app=self.vendor,
+            tracer=tracer,
+            service=self.service,
+            app_type=sqlx.APP_TYPE,
+        ).onto(engine)
+
         listen(engine, 'before_cursor_execute', self._before_cur_exec)
         listen(engine, 'after_cursor_execute', self._after_cur_exec)
         listen(engine, 'dbapi_error', self._dbapi_error)
 
     def _before_cur_exec(self, conn, cursor, statement, *args):
-        span = self.tracer.trace(
+        pin = Pin.get_from(self.engine)
+        if not pin or not pin.enabled():
+            # don't trace the execution
+            return
+
+        span = pin.tracer.trace(
             self.name,
-            service=self.service,
+            service=pin.service,
             span_type=sqlx.TYPE,
-            resource=statement)
+            resource=statement,
+        )
 
         if not _set_tags_from_url(span, conn.engine.url):
             _set_tags_from_cursor(span, self.vendor, cursor)
 
     def _after_cur_exec(self, conn, cursor, statement, *args):
-        span = self.tracer.current_span()
+        pin = Pin.get_from(self.engine)
+        if not pin or not pin.enabled():
+            # don't trace the execution
+            return
+
+        span = pin.tracer.current_span()
         if not span:
             return
 
         try:
             if cursor and cursor.rowcount >= 0:
                 span.set_tag(sqlx.ROWS, cursor.rowcount)
+            elif cursor and cursor.rowcount == -1:
```

Comment:
From the doc:
https://docs.python.org/3.0/library/sqlite3.html#sqlite3.Cursor.rowcount

> As required by the Python DB API Spec, the rowcount attribute “is -1 in case no executeXX() has been performed on the cursor or the rowcount of the last operation is not determinable by the interface”.

So it is possible that multiple rows got affected, but the client can't know.

So, instead of putting something arbitrarily null, we could just not set `sql.rows` when we have `-1`?